cluster_management/src/main/java/com/pinterest/rocksplicator/ConfigGenerator.java

```diff
@@ -51,20 +51,23 @@
 import java.util.Map;
 import java.util.Set;
 import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
 
 public class ConfigGenerator extends RoutingTableProvider implements CustomCodeCallbackHandler {
 
   private static final Logger LOG = LoggerFactory.getLogger(ConfigGenerator.class);
 
   private final String clusterName;
-  private HelixManager helixManager;
-  private Map<String, String> hostToHostWithDomain;
   private final String postUrl;
-  private JSONObject dataParameters;
-  private String lastPostedContent;
-  private Set<String> disabledHosts;
-  private RocksplicatorMonitor monitor;
   private final boolean enableDumpToLocal;
+  private final Map<String, String> hostToHostWithDomain;
+  private final JSONObject dataParameters;
+  private final RocksplicatorMonitor monitor;
+  private final ReentrantLock synchronizedCallbackLock;
+  private HelixManager helixManager;
+  private volatile String lastPostedContent;
+  private volatile Set<String> disabledHosts;
```

Comment:
why volatile?
Previously the code is under the assumption that it is single threaded.
Now on top of that, we have a global lock for all the public interfaces of the class.

Reply:
Not needed.